### PR TITLE
fix(loans): route a request to the waitlist when no copy is assignable now (#384)

### PR DESCRIPTION
**Pull request by fabiodalez-dev**
Follow-up to #385 (the graceful-400 approval fix): completes the #384 behaviour so a request on a book already on loan becomes a **real reservation**, not a copy-less pending loan that bypasses the FIFO queue and cannot be approved while the copy is out.

## Behaviour (`createReservation`, the unified public calendar submit)

Decided per physical availability, after the canonical book lock:
- **physical copies but none assignable for the window** (`hasAssignableInLibraryCopyThrough`) → create a real `prenotazioni.attiva` waitlist row (`createActiveReservation`, race-safe `queue_position`), return `status='reserved'`;
- **a copy is genuinely assignable** → normal loan-request / auto-approve flow;
- **legacy aggregate-only titles** (no `copie` rows) → keep the pending-loan fallback (the promotion pipeline can never bind a physical copy).

`hasAssignableInLibraryCopyThrough(bookId, endDate)` — an in-library copy (`disponibile`/`prenotato`) with **no commitment starting on or before the requested end** (`data_prestito <= endDate`):
- `prenotato`/`da_ritirare`/pending-with-copy committed **before** the request → reservation;
- a commitment starting **after** the requested end (incl. exactly `end+1`) → loan allowed;
- a `prestato` copy → reservation (a contractual due date is not a guaranteed return);
- multi-copy → the selector explicitly picks the copy free of a prior-return dependency.

Also refines the admin **cancel-pickup** (#381): a voluntary cancellation on/before the deadline records `annullato`, only an elapsed deadline records `scaduto`; the default reason / email no longer assert a missed pickup. `book-detail.php` surfaces the reserved outcome with its own SweetAlert copy.

## Existing installations
**Pure code — no schema or migration change.** The `prenotazioni` columns used are canonical and already relied on by `reserve()`, so existing installs get the behaviour on the code update; pre-existing copy-less pending loans degrade gracefully (400, not 500) and remain admin-manageable.

## Validation
50 targeted #301/#384 checks, 66 loan edge-case checks, reservation-consistency + CodeRabbit-regression tests, PHPStan level 5 clean, i18n parity green. `loan-auto-approval-301` + `plugin-audit-hardening-regressions` also re-run locally against the dev DB (with existing data).

## Summary by CodeRabbit

- **Nuove funzionalità**
  - Le richieste non assegnabili vengono inserite in una lista d’attesa FIFO.
  - L’assegnazione privilegia le copie senza conflitti con impegni futuri.
  - La conferma distingue tra prestito e prenotazione in coda, mostrando intervallo e stato.

- **Correzioni**
  - La cancellazione del ritiro distingue correttamente annullamento volontario e scadenza.
  - Aggiornati motivi e notifiche per ritiri annullati o scaduti.
  - Migliorata la verifica della disponibilità nelle richieste automatiche e manuali.

- **Localizzazione**
  - Aggiunte traduzioni per le prenotazioni in italiano, inglese, tedesco, francese e danese.

- **Test**
  - Ampliate le verifiche su prenotazioni, cancellazioni e assegnazione delle copie.

### CODE REVIEW

File: tests/loan-pickup-cancelled-softdelete.unit.php [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    public bool $localeResolved = false;
+    public string $capturedEmail = '';
+
+    public function resolveRecipientLocale(string $email): string
+    {
+        $this->localeResolved = true;
+        $this->capturedEmail = $email;
+        return 'it_IT';
+    }
+}
```
[the remark this answers is not in the document]
Comment (fabiodalez-dev) [reply tests/loan-pickup-cancelled-softdelete.unit.php:67]: Applied in 00b9e538: the overridden resolveRecipientLocale() now throws right after recording the probe flag, so the JOIN-reached assertions still hold but the method returns before sendWithRetry() can touch the mail transport. No real send happens during the unit run.

**Comment by fabiodalez-dev** [comment]
Reopening immediately to retrigger GitHub Actions for the latest Mobile API and timezone-fixture commit; the synchronize event did not create workflow runs.
